sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpAnnotatedMessageExtensions.cs

```diff
@@ -215,7 +221,13 @@ public static DateTimeOffset GetEnqueuedTime(this AmqpAnnotatedMessage instance,
             if ((instance.HasSection(AmqpMessageSection.MessageAnnotations))
                 && (instance.MessageAnnotations.TryGetValue(AmqpProperty.EnqueuedTime.ToString(), out var value)))
             {
-                return (DateTimeOffset)value;
+                return value switch
+                {
+                    DateTime dateValue => new DateTimeOffset(dateValue, TimeSpan.Zero),
+                    long longValue => new DateTimeOffset(longValue, TimeSpan.Zero),
```

Comment:
dumb q but does this ever get hit? Wouldn't the previous code have thrown if this value was a long?

Reply:
No, because this logic was in a "normalize" step embedded in the converter that mutated values in the source `AmqpMessage` instance before we read it.   So we always read a consistent and normalized type.